include/onnxruntime/core/framework/op_kernel.h

```diff
@@ -49,6 +49,26 @@ class OpKernel {
     ORT_NOT_IMPLEMENTED(__FUNCTION__, " is not implemented");
   }
 
+  // Override this function to PrePack the constant Tensor to the format as needed.
+  // For example, MatMul kernel can pack the input B if it is constant like code below.
+  //   Status PrePack(const Tensor& tensor, int input_idx, bool& is_packed) override {
+  //     is_packed = false;
+  //     if (input_idx == 1) {
+  //       this.Pack(tensor, this.buffer_);
+  //     }
+  //     return Status::OK();
+  //   }
+
```

Comment:
nit: probably need a more complete example showing that the kernel needs to do an allocation and have a member that owns that buffer.